subsys/net/lib/nrf_cloud/include/nrf_cloud_codec.h

```diff
@@ -37,7 +37,8 @@ int nrf_cloud_decode_requested_state(const struct nrf_cloud_data *payload,
 /**@brief Decodes data endpoint information. */
 int nrf_cloud_decode_data_endpoint(const struct nrf_cloud_data *input,
 				   struct nrf_cloud_data *tx_endpoint,
-				   struct nrf_cloud_data *rx_endpoint);
+				   struct nrf_cloud_data *rx_endpoint,
+				   struct nrf_cloud_data *endpoint);
```

Comment:
So this new `endpoint` - is it only related to PIN association or is it something more? Perhaps we could use a more precise name for it? `endpoint` is pretty ambiguous and hard to follow in the code below.

Reply:
It's supposed to replace both rx and tx endpoints in the future, so there will be only one endpoint. For now, this is only future-proofing, as more changes are required both on the cloud and device end of things.

Reply:
I'll have another look at the naming

Reply:
Called it `m_endpoint`, as that's what it's called on the AWS side of things

Reply:
Looks dangerously like Hungarian notation, but if it's mirroring the AWS I think it's fine.